pkg/config/patch.go

```diff
@@ -72,10 +72,6 @@ func ApplyPatches[K any](object, out K, patches []v1alpha1.Patch) (int, bool, er
 					errs = append(errs, fmt.Errorf("error applying patch %d, to object: %w", i, err))
 					continue
 				}
-				if err := json.Unmarshal(patched, out); err != nil {
```

Comment:
Unmarshalling the "intermediate" patched versions over the same object is the cause of the bug. This isn't necessary though, we unmarshal the whole object after all patches are applied.